File: M02/Assignment/SprintChallenge3_PokeApi/app/src/main/java/com/lambdaschool/sprintchallenge3_pokeapi/MainActivity.java
package com.lambdaschool.sprintchallenge3_pokeapi;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    LinearLayout namesLayout;
    Context context;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        context = this;
        namesLayout = findViewById(R.id.names_list_layout);

        findViewById(R.id.search_button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(context, PokemonDetailsActivity.class);
                intent.putExtra("Search_Parameter", ((EditText)findViewById(R.id.search_bar)).getText().toString());
                startActivityForResult(intent, 0);
            }
        });
    }

    private TextView buildTextView(final String name) {
        TextView view = new TextView(this);
        view.setText(name);
        view.setTextSize(14);
        view.setPadding(10, 10, 10, 10);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(context, PokemonDetailsActivity.class);
                intent.putExtra("Search_Parameter", name);
                startActivityForResult(intent, 0);
            }
        });
        view.setOnLongClickListener(new View.OnLongClickListener() {
            @Override
            public boolean onLongClick(View view) {
                namesLayout.removeView(view);
                return true;
            }
        });
        return view;
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        if(resultCode == Activity.RESULT_OK) {
            switch (requestCode) {
                case 0:
                    final Pokemon pokemon = (Pokemon) data.getSerializableExtra("pokemon");
                    if(pokemon != null) {
                        namesLayout.addView(buildTextView(pokemon.getName()));
                    }
                    break;
            }
        }
    }
}

File: M02/Assignment/SprintChallenge3_PokeApi/app/src/main/java/com/lambdaschool/sprintchallenge3_pokeapi/MainActivity.kt
package com.lambdaschool.sprintchallenge3_pokeapi

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.LinearLayout
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    internal var namesLayout: LinearLayout? = null
    internal var context: Context? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        context = this
        namesLayout = findViewById(R.id.names_list_layout)

        findViewById<View>(R.id.search_button).setOnClickListener {
            val intent = Intent(context, PokemonDetailsActivity::class.java)
            intent.putExtra("Search_Parameter", (findViewById<View>(R.id.search_bar) as EditText).text.toString())
            startActivityForResult(intent, 0)
        }
    }

    private fun buildTextView(name: String?): TextView {
        val view = TextView(this)
        view.text = name
        view.textSize = 14f
        view.setPadding(10, 10, 10, 10)
        view.setOnClickListener {
            val intent = Intent(context, PokemonDetailsActivity::class.java)
            intent.putExtra("Search_Parameter", name)
            startActivityForResult(intent, 0)
        }
        view.setOnLongClickListener { clickedView ->
            namesLayout?.removeView(clickedView)
            true
        }
        return view
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        if (resultCode == Activity.RESULT_OK) {
            when (requestCode) {
                0 -> {
                    val pokemon = data!!.getSerializableExtra("pokemon") as Pokemon
                    namesLayout?.addView(buildTextView(pokemon.name))
                }
            }
        }
    }
}
